app/models/bulk_submission.rb

```diff
@@ -224,24 +225,24 @@ def submission_structure
       csv_data_rows.each_with_index do |row, index|
         next if row.all?(&:nil?)
 
-        details = Hash[headers.each_with_index.map do |header, pos|
-                         validate_entry(header, pos, row, index + start_row)
-                       end ].merge('row' => index + start_row)
+        details = headers.each_with_index.filter_map do |header, pos|
+          validate_entry(header, pos, row, index + start_row)
+        end.to_h.merge('row' => index + start_row)
```

Comment:
Related to above comment - not quite sure if the new version is equivalent to the old?